tools/pubsub2inbox/output/mail.py

```diff
@@ -0,0 +1,380 @@
+        if 'to' not in self.output_config:
+            raise Exception('No recipient (to) configured for email output!')
+        if 'subject' not in self.output_config:
+            raise Exception('No subject configured for email output!')
+
+        if 'body' not in self.output_config:
+            raise Exception('No body configured for email output!')
+
+        for mail_type in ['html', 'text']:
+            if mail_type in self.output_config['body']:
+                mail_template = self.jinja_environment.from_string(
+                    self.output_config['body'][mail_type])
+                mail_template.name = mail_type
+                mail['%s_body' % mail_type] = mail_template.render()
+
+        if mail['html_body'] == '' and mail['text_body'] == '':
+            raise Exception(
+                'No HMTL or text email body configured for email output!')
+
+        for tpl in ['from', 'to', 'subject']:
+            mail_template = self.jinja_environment.from_string(
+                self.output_config[tpl])
+            mail['mail_%s' % tpl] = mail_template.render()
+
+        self.logger.debug('Canonicalizing email formats...')
+        # Canonicalize the email formats
+        for tpl in ['from', 'to']:
+            parsed_emails = email.utils.getaddresses([mail['mail_%s' % tpl]])
+            if tpl == 'from' and len(parsed_emails) > 1:
+                raise Exception('Multiple senders in from field!')
+            new_email = ''
+            for e in parsed_emails:
+                new_email += ', ' if new_email != '' else ''
+                new_email += email.utils.formataddr(e)
+            mail['mail_%s' % tpl] = new_email
+
+        if 'expandGroupRecipients' in self.output_config and self.output_config[
+                'expandGroupRecipients']:
+
+            to_emails = email.utils.getaddresses([mail['mail_to']])
+            self.logger.debug('Starting expansion of group recipients...',
+                              extra={'to': to_emails})
+
+            user_credentials = Credentials(
+                self.get_token_for_scopes([
+                    'https://www.googleapis.com/auth/admin.directory.user.readonly'
+                ],
+                                          service_account=self.
+                                          output_config['serviceAccountEmail']
+                                          if 'serviceAccountEmail'
+                                          in self.output_config else None))
+            group_credentials = Credentials(
+                self.get_token_for_scopes([
+                    'https://www.googleapis.com/auth/cloud-identity.groups.readonly'
+                ],
+                                          service_account=self.
+                                          output_config['serviceAccountEmail']
+                                          if 'serviceAccountEmail'
+                                          in self.output_config else None))
```

Comment:
Slightly confusing formatting of code. You could create a `service_account` variable and simplify the creation of the `Credentials` objects, so the code would be more legible.